pkg/landscaper/installations/subinstallations/subinstallations.go

```diff
@@ -151,7 +153,13 @@ func (o *Operation) createOrUpdateNewInstallation(ctx context.Context, inst *lsv
 
 	if subInst == nil {
 		subInst = &lsv1alpha1.Installation{}
-		subInst.GenerateName = fmt.Sprintf("%s-", subInstTmpl.Name)
+
+		generateName := subInstTmpl.Name
+		if len(generateName) > validation.InstallationGenerateNameMaxLength-1 {
+			generateName = generateName[:validation.InstallationGenerateNameMaxLength-1]
+		}
+
+		subInst.GenerateName = fmt.Sprintf("%s-", generateName)
 		subInst.Namespace = inst.Namespace
```

Comment:
here is the modification of the generateName

Reply:
this cuts of too long names. 
I meant that we could also add the same restriction to installation templates as for installations.
So we could add 
```
if len(objMeta.GetName()) > InstallationNameMaxLength {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), objMeta.GetName(), validation.MaxLenError(InstallationNameMaxLength)))
	} else if len(objMeta.GetGenerateName()) > InstallationGenerateNameMaxLength {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("generateName"), objMeta.GetGenerateName(), validation.MaxLenError(InstallationGenerateNameMaxLength)))
	}
```
 to https://github.com/gardener/landscaper/blob/master/pkg/apis/core/validation/blueprint.go#L262